Fix parallel builds: subdirectories must be created sequentially and gmake-4.1
fails to do that. Unfortunately, gmake-4.1 does not know about .WAIT, so one
must fake it, which should not hurt for implementations knowing about .WAIT.

--- src/lib/omnithread/dir.mk~	2023-09-25 22:15:42.000000000 +0200
+++ src/lib/omnithread/dir.mk	2024-06-21 11:22:46.254916333 +0200
@@ -59,10 +59,12 @@
 staticlib := static/$(patsubst %,$(LibNoDebugPattern),$(LIB_NAME)$(major))
 MDFLAGS += -p static/
 
-mkstatic::
+.WAIT:
+.NOTPARALLEL: .WAIT
+mkdir::
 	@(dir=static; $(CreateDir))
 
-mkstatic:: $(staticlib)
+mkstatic:: mkdir .WAIT $(staticlib)
 
 $(staticlib): $(patsubst %, static/%, $(LIB_OBJS))
 	@$(StaticLinkLibrary)
@@ -108,10 +108,10 @@
   imps := $(LIB_IMPORTS)
 endif
 
-mkshared::
+mkdir::
 	@(dir=shared; $(CreateDir))
 
-mkshared:: $(shlib)
+mkshared:: mkdir .WAIT $(shlib)
 
 $(shlib): $(patsubst %, shared/%, $(LIB_OBJS))
 	@(namespec="$(namespec)" extralibs="$(imps) $(extralibs)" nodeffile=1; \
--- mk/mklib.mk~	2009-07-02 15:24:15.000000000 +0200
+++ mk/mklib.mk	2024-06-24 11:55:35.088327309 +0200
@@ -43,10 +43,12 @@
 staticlib := static/$(patsubst %,$(LibNoDebugPattern),$(LIB_NAME)$(major))
 MDFLAGS += -p static/
 
-mkstatic::
+.WAIT:
+.NOTPARALLEL: .WAIT
+mkdir::
 	@(dir=static; $(CreateDir))
 
-mkstatic:: $(staticlib)
+mkstatic:: mkdir .WAIT $(staticlib)
 
 $(staticlib): $(patsubst %, static/%, $(LIB_OBJS))
 	@$(StaticLinkLibrary)
@@ -91,10 +93,10 @@
   imps := $(LIB_IMPORTS)
 endif
 
-mkshared::
+mkdir::
 	@(dir=shared; $(CreateDir))
 
-mkshared:: $(shlib)
+mkshared:: mkdir .WAIT $(shlib)
 
 $(shlib): $(patsubst %, shared/%, $(LIB_OBJS) $(LIB_SHARED_ONLY_OBJS))
 	@(namespec="$(namespec)" extralibs="$(imps) $(extralibs)"; \
--- src/services/mklib/stublib/dir.mk~	2012-04-14 11:48:18.000000000 +0200
+++ src/services/mklib/stublib/dir.mk	2024-06-27 11:38:04.891724529 +0200
@@ -33,10 +33,12 @@
 sk = static/$(patsubst %,$(LibNoDebugPattern),$(COS_SKLIB_NAME)$(version))
 MDFLAGS += -p static/
 
-mkstatic::
+.WAIT:
+.NOTPARALLEL: .WAIT
+mkdir::
 	@(dir=static; $(CreateDir))
 
-mkstatic:: $(sk)
+mkstatic:: mkdir .WAIT $(sk)
 
 $(sk): $(patsubst %, static/%, $(COS_SK_OBJS))
 	@$(StaticLinkLibrary)
@@ -80,10 +82,10 @@
 imps := $(OMNIORB_LIB)
 endif
 
-mkshared::
+mkdir::
 	@(dir=shared; $(CreateDir))
 
-mkshared:: $(skshared)
+mkshared:: mkdir .WAIT $(skshared)
 
 $(skshared): $(patsubst %, shared/%, $(COS_SK_OBJS))
 	@(namespec="$(sknamespec)"; extralibs="$(imps) $(extralibs)"; \
--- src/services/mklib/dynstublib/dir.mk~	2018-09-05 17:21:03.000000000 +0200
+++ src/services/mklib/dynstublib/dir.mk	2024-06-27 11:39:53.556523074 +0200
@@ -33,10 +33,12 @@
 dynsk = static/$(patsubst %,$(LibNoDebugPattern),$(COS_DYNSKLIB_NAME)$(version))
 MDFLAGS += -p static/
 
-mkstatic::
+.WAIT:
+.NOTPARALLEL: .WAIT
+mkdir::
 	@(dir=static; $(CreateDir))
 
-mkstatic:: $(dynsk)
+mkstatic:: mkdir .WAIT $(dynsk)
 
 $(dynsk): $(patsubst %, static/%, $(COS_DYNSK_OBJS))
 	@$(StaticLinkLibrary)
@@ -80,10 +82,10 @@
 dynimps := $(OMNIORB_COS_LIB) $(OMNIORB_LIB)
 endif
 
-mkshared::
+mkdir::
 	@(dir=shared; $(CreateDir))
 
-mkshared:: $(dynskshared) 
+mkshared:: mkdir .WAIT $(dynskshared) 
 
 $(dynskshared): $(skshared) $(patsubst %, shared/%, $(COS_DYNSK_OBJS))
 	@(namespec="$(dynsknamespec)"; extralibs="$(dynimps)"; \
